lib/kafka_ex/consumer_group/manager.ex

```diff
@@ -280,6 +280,10 @@ defmodule KafkaEx.ConsumerGroup.Manager do
 
     case sync_group_response do
       %SyncGroupResponse{error_code: :no_error, assignments: assignments} ->
+        # On a high-latency connection, the join/sync process takes a long
+        # time. Send a heartbeat as soon as possible to avoid hitting the
+        # session timeout.
+        send(self(), :heartbeat)
```

Comment:
This won't execute the heartbeat until this function is finished - `send` will put a message in the mailbox but the mailbox isn't addressed until the current thing is finished. Was that your intention or did you intend to synchronously send a sync now?

Reply:
we still need to finish and update the state before processing the heartbeat on the next loop cycle, I think?

Reply:
Just making sure this is what you meant to do.  I wonder if there's a place to put it in the code so that it's synchronous and you can see exactly where it's happening?

Reply:
ok, I made it synchronous